tests/widgets.js

```diff
@@ -0,0 +1,93 @@
+require([
+    'chai',
+    'jquery',
+    'underscore',
+    'tests/utils',
+    'vellum/expressionEditor',
+    'vellum/util'
+], function (
+    chai,
+    $,
+    _,
+    util,
+    expressionEditor,
+    vellumUtil
+) {
+    var assert = chai.assert,
+        real_showXPathEditor = expressionEditor.showXPathEditor,
+        events = {};
+
+    vellumUtil.eventuality(events);
+
+    // setup event so we can hook into xpath editor show
+    expressionEditor.showXPathEditor = function () {
+        real_showXPathEditor.apply(null, arguments);
+        events.fire("showXPathEditor");
+    };
+
+    describe("The widgets module", function () {
+        before(function (done) {
+            util.init({
+                javaRosa: { langs: ['en'] },
+                core: {
+                    onReady: function () {
+                        assert(this.isPluginEnabled("modeliteration"),
+                               "modeliteration plugin should be enabled");
+                        done();
+                    }
+                }
+            });
+        });
+
+        it("xPath widget should show newlines in advanced mode", function (done) {
+            // HACK this ugly test references a lot of implementation details
+            util.loadXML("");
+            var value = '"line 1 \n line 2"',
+                escaped = '"line 1 &#10; line 2"';
+            util.addQuestion("DataBindOnly", "hidden", {
+                calculateAttr: value
+            });
+            util.clickQuestion("/data/hidden");
+            var input = $("[name=property-calculateAttr]");
+            assert.equal(input.val(), escaped);
+
+            // click Edit button
+            input.closest("div").siblings(".fd-edit-button").click();
```

Comment:
Picky, but could this be `input.closest(".control-group").find(".fd-edit-button")`?  I think both looking for a generic ancestor div, and using siblings, are kind of fragile.